src/zino/trapobservers/link_traps.py

```diff
@@ -0,0 +1,192 @@
+        # but we have no more Cisco devices to test on, so this has been deliberately left out.
+
+        is_up = trap.name == "linkUp"
+        self.handle_link_transition(trap.agent.device, port, is_up)
+
+    def handle_link_transition(self, device: DeviceState, port: Port, is_up: bool, reason: Optional[str] = None):
+        """This method is called when a linkUp or linkDown trap is received, and updates event state accordingly -
+        including the logic for handling flapping states.
+        """
+        if self.is_port_ignored_by_patterns(device, port.ifdescr):
+            return
+
+        # At this point, legacy Zino would ignore so-called "insignificant" interfaces.  As mentioned in the
+        # LinkStateTask docstring, this concept has deliberately not been ported to Zino 2.
+
+        _logger.debug(
+            "%s: intf %s ix %d link%s%s",
+            device.name,
+            port.ifdescr,
+            port.ifindex,
+            "Up" if is_up else "Down",
+            f", {reason}" if reason else "",
+        )
+
+        if self.is_port_ignored_by_policy(device, port, is_up):
+            return
+
+        index = (device.name, port.ifindex)
+        self.update_interface_flapping_score(index)
+
+        new_state = InterfaceState.UP if is_up else InterfaceState.DOWN
+
+        if self.is_interface_flapping(index):
+            # TODO: if event doesn't exist, create it
+            # TODO: Record new number of flaps in event
+            # TODO: When flapcount modulo 100 is zero, log a message with flapping stats
+            pass
+        else:
+            event: PortStateEvent = self.state.events.get_or_create_event(device.name, port.ifindex, PortStateEvent)
+
+            event.portstate = new_state
+            event.port = port.ifdescr
+            event.ifindex = port.ifindex
+            port.state = new_state
+            if polldev := self.polldevs.get(device.name):
+                event.polladdr = polldev.address
+                event.priority = polldev.priority
+            event.descr = port.ifalias  # or value received from trap? see ldescr from legacy Zino
+
+            # TODO: If there is internal flapping state, log it in the event and clear internal state
+            # TODO: Set final flapcount in event
+
+            msg = f'{device.name}: intf "{port.ifdescr}" ix {port.ifindex} link{new_state.capitalize()}'
+            if reason:
+                event.reason = reason
+                msg = f'{msg}, "{reason}"'
+            _logger.info(msg)
+            event.add_log(msg)
+            self.state.events.commit(event)
+
+            poll = LinkStateTask(device=polldev, state=self.state)
+            poll.schedule_verification_of_single_port(port.ifindex, deadline=timedelta(seconds=0), reason="trap-verify")
+            poll.schedule_verification_of_single_port(
+                port.ifindex, deadline=timedelta(minutes=2), reason="trap-verify-2"
+            )
```

Comment:
Even tho these timedeltas are only used once, it might still be good to make them constants (I have suggested constants quite a bit now so might get cluttered at the top of the file :stuck_out_tongue: )

Reply:
The two-minute deadline could do as a constant, but the 0 second one is just sort of a "workaround" for the fact that this isn't an async function, and we want to call an async function in a  "fire and forget" manner:   This function doesn't care about the results of the poll operation - it just needs to be scheduled be run *once*, as soon as the event loop resumes.